File: java-hfsframework-primefaces/hfsframework-admin/src/main/java/br/com/hfsframework/admin/AdmFuncionalidadeBC.java
/**
 * <p><b>HFS Framework</b></p>
 * @author Henrique Figueiredo de Souza
 * @version 1.0
 * @since 2017
 */
package br.com.hfsframework.admin;

import br.com.hfsframework.admin.data.AdmFuncionalidadeRepository;
import br.com.hfsframework.admin.model.AdmFuncionalidade;
import br.com.hfsframework.base.BaseBusinessController;

/**
 * The Class AdmFuncionalidadeBC.
 */
public class AdmFuncionalidadeBC extends BaseBusinessController<AdmFuncionalidade, Long, AdmFuncionalidadeRepository> {

	/** The Constant serialVersionUID. */
	private static final long serialVersionUID = 1L;

	/* (non-Javadoc)
	 * @see br.com.hfsframework.base.BaseBusinessController#getDescricao(java.io.Serializable)
	 */
	@Override
	public String getDescricao(Long id){
		return repositorio.getDescricao(id);
	}
	
	/* (non-Javadoc)
	 * @see br.com.hfsframework.base.BaseBusinessController#existeNovo(java.lang.String)
	 */
	@Override
	public boolean existeNovo(String novo){
		Long existe = repositorio.countNovo(novo.toLowerCase());
		return (existe > 0);
	}
	
	/* (non-Javadoc)
	 * @see br.com.hfsframework.base.BaseBusinessController#existeAntigoNovo(java.lang.String, java.lang.String)
	 */
	@Override
	public boolean existeAntigo(Long id, String novo){
		String antigo = getDescricao(id);
		Long existe = repositorio.countAntigo(antigo.toLowerCase(), novo.toLowerCase());
		return (existe > 0);
	}
	

}

File: csharp-hfsframework/hfsframework-admin/hfsframework.admin/AdmFuncionalidadeBC.cs
using hfsframework.hfsbase;
using hfsframework.admin.model;
using hfsframework.admin.data;

namespace hfsframework.admin
{
    public class AdmFuncionalidadeBC: BaseBusinessController<AdmFuncionalidade, long, AdmFuncionalidadeRepository>
    {
        private static AdmFuncionalidadeBC instancia = new AdmFuncionalidadeBC();

        public static AdmFuncionalidadeBC Instancia
        {
            get
            {
                return instancia;
            }
        }

        private AdmFuncionalidadeBC():base(new AdmFuncionalidadeRepository())
        {
        }
    }
}
